packages/typescript-estree/src/convert.ts

```diff
@@ -2407,36 +2431,40 @@ export class Converter {
         }
         return result;
       }
-      case SyntaxKind.ConstructorType:
+      case SyntaxKind.ConstructorType: {
+        const result = this.createNode<TSESTree.TSConstructorType>(node, {
+          type: AST_NODE_TYPES.TSConstructorType,
+          params: this.convertParameters(node.parameters),
+          abstract: hasModifier(SyntaxKind.AbstractKeyword, node),
+        });
+        if (node.type) {
+          result.returnType = this.convertTypeAnnotation(node.type, node);
+        }
+        if (node.typeParameters) {
+          result.typeParameters = this.convertTSTypeParametersToTypeParametersDeclaration(
+            node.typeParameters,
+          );
+        }
+        return result;
+      }
+
       case SyntaxKind.FunctionType:
       case SyntaxKind.ConstructSignature:
       case SyntaxKind.CallSignature: {
-        let type: AST_NODE_TYPES;
-        switch (node.kind) {
-          case SyntaxKind.ConstructSignature:
-            type = AST_NODE_TYPES.TSConstructSignatureDeclaration;
-            break;
-          case SyntaxKind.CallSignature:
-            type = AST_NODE_TYPES.TSCallSignatureDeclaration;
-            break;
-          case SyntaxKind.FunctionType:
-            type = AST_NODE_TYPES.TSFunctionType;
-            break;
-          case SyntaxKind.ConstructorType:
-          default:
-            type = AST_NODE_TYPES.TSConstructorType;
-            break;
-        }
+        const type =
+          node.kind === SyntaxKind.ConstructSignature
+            ? AST_NODE_TYPES.TSConstructSignatureDeclaration
+            : node.kind === SyntaxKind.CallSignature
+            ? AST_NODE_TYPES.TSCallSignatureDeclaration
+            : AST_NODE_TYPES.TSFunctionType;
```

Comment:
IMO nested ternaries are awful for readability.
If your intention is to make `type` a `const`, I'd suggest something like an IIFE.

```suggestion
        const type = ((): AST_NODE_TYPES => {
          switch (node.kind) {
            case SyntaxKind.ConstructSignature:
              return AST_NODE_TYPES.TSConstructSignatureDeclaration;
              
            case SyntaxKind.CallSignature:
              return AST_NODE_TYPES.TSCallSignatureDeclaration;
              
            case SyntaxKind.FunctionType:
              return AST_NODE_TYPES.TSFunctionType;
              
            case SyntaxKind.ConstructorType:
            default:
              return AST_NODE_TYPES.TSConstructorType;
          }
        })();
```

Reply:
yes and no, bdw, nowthis case only handles:
`SyntaxKind.FunctionType`, `SyntaxKind.ConstructSignature`, `SyntaxKind.CallSignature` 

as `ConstructorType` is handled separately

Reply:
if you really want we can change it to
```suggestion
        let type:
          | AST_NODE_TYPES.TSConstructSignatureDeclaration
          | AST_NODE_TYPES.TSCallSignatureDeclaration
          | AST_NODE_TYPES.TSFunctionType;
        switch (node.kind) {
          case SyntaxKind.ConstructSignature:
            type = AST_NODE_TYPES.TSConstructSignatureDeclaration;
            break;
          case SyntaxKind.CallSignature:
            type = AST_NODE_TYPES.TSCallSignatureDeclaration;
            break;
          case SyntaxKind.FunctionType:
          default:
            type = AST_NODE_TYPES.TSFunctionType;
            break;
        }
```

Reply:
it was just a suggestion if you wanted to do it during the master merge - I'm happy to merge this as is.